src/loggers/ProgramLogger.ts

```diff
@@ -44,14 +47,31 @@ export class ProgramLogger implements Logger<string> {
         return this.log(content, "Error");
     }
 
-    private log(content: string, messageType: MessageType): Promise<void> {
+    batchLog(content: BatchLogContent): Promise<void> {
         if (this.loggerRepository) {
             const options = { isDebug: this.isDebug };
-            return new LogMessageUseCase(this.loggerRepository)
-                .execute({ message: content, messageType: messageType }, options)
+            const logs = content.map(content =>
+                this.mapContentToLog(content.content, content.messageType)
+            );
+            return new BatchLogMessageUseCase(this.loggerRepository)
+                .execute(logs, options)
                 .toPromise();
         } else {
             throw new Error(`Logger not initialized properly. Please check configuration.`);
         }
     }
+
+    private log(content: string, messageType: MessageType): Promise<void> {
+        if (this.loggerRepository) {
+            const options = { isDebug: this.isDebug };
+            const log = this.mapContentToLog(content, messageType);
+            return new LogMessageUseCase(this.loggerRepository).execute(log, options).toPromise();
+        } else {
+            throw new Error(`Logger not initialized properly. Please check configuration.`);
+        }
+    }
+
+    private mapContentToLog(content: string, messageType: MessageType): DefaultLog {
+        return { message: content, messageType: messageType };
+    }
```

Comment:
ConsoleLogger and ProgramLogger have some repeated code. Unless this is accidental repetition, they should be DRYed with some common code.